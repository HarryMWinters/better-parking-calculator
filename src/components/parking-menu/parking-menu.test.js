import React from "react";
import renderer from "react-test-renderer";

import ParkingMenu from "./parking-menu";

const parkingInfo = {
  "Economy Parking": {
    "Per Hour": 2.0,
    "Daily Maximum": 9.0,
    "Per Week": 63.0
  },
  "Short-Term Parking": {
    "Per Hour": 2.0,
    "Daily Maximum": 24.0,
    "Per Week": 168.0
  },
  "Long-Term Surface Parking": {
    "Per Hour": 2.0,
    "Daily Maximum": 12.0,
    "Per Week": 84.0
  },
  "Long-Term Garage Parking": {
    "Per Hour": 2.0,
    "Daily Maximum": 17.0,
    "Per Week": 119.0
  },
  "Valet Parking": {
    "Per Hour": null,
    // Valet parking is only charged per day.
    "Daily Maximum": 24.0,
    "Per Week": 169.0
  }
};

it("Matches snapshot", () => {
  const tree = renderer
    .create(<ParkingMenu parkingInfo={parkingInfo} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
