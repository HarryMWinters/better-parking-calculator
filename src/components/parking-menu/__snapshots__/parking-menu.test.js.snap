// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div>
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th />
        <th>
          Per Hour
        </th>
        <th>
          Daily Maximum
        </th>
        <th>
          Per Week
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Economy Parking
        </td>
        <td>
          $
          2
          .00
        </td>
        <td>
          $
          9
          .00
        </td>
        <td>
          $
          63
          .00
        </td>
      </tr>
      <tr>
        <td>
          Short-Term Parking
        </td>
        <td>
          $
          2
          .00
        </td>
        <td>
          $
          24
          .00
        </td>
        <td>
          $
          168
          .00
        </td>
      </tr>
      <tr>
        <td>
          Long-Term Surface Parking
        </td>
        <td>
          $
          2
          .00
        </td>
        <td>
          $
          12
          .00
        </td>
        <td>
          $
          84
          .00
        </td>
      </tr>
      <tr>
        <td>
          Long-Term Garage Parking
        </td>
        <td>
          $
          2
          .00
        </td>
        <td>
          $
          17
          .00
        </td>
        <td>
          $
          119
          .00
        </td>
      </tr>
      <tr>
        <td>
          Valet Parking
        </td>
        <td>
          N/A
        </td>
        <td>
          $
          24
          .00
        </td>
        <td>
          $
          169
          .00
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
